DOAX-VenusVacation/d3dx.ini

```diff
@@ -1594,6 +1836,64 @@ if $shadow_mode == 2
 	handling = skip
 endif
 
+[ShaderOverrideTanPS]
+; NOTE: Only executed when the tan textures are updated, e.g.
+; by clicking on the sunscreen buttons.
+hash = c65ebfdf7e4e8f50
+allow_duplicate_hash=overrule
+$shader_type = 6
+if $costume_mods
+  checktextureoverride = ps-t0
+endif
+
+[ShaderOverrideWater-Old]
+; Moving Water
+; Bytecode hash: 4b4b431a
+hash = ea320288619c5ee6
+run = CommandListEnvironment
+[ShaderOverrideWater-20200610]
+; Bytecode hash: 19eb5ff4
+hash = 0f0a2c40c629c533
+run = CommandListEnvironment
+
+[ShaderOverrideVenusCard-Old]
+; Bytecode hash: ?
+hash = 022ead00ab05e856
+run = CommandListEnvironment
+[ShaderOverrideVenusCard-20200610]
+; Bytecode hash: 3be88682
+hash = 892f7246173bea98
+run = CommandListEnvironment
+
+; Palm Tree Leafs
+[ShaderOverridePalmTrees-Old]
+; Bytecode hash: edfcd195
+hash = b9e5304ff1a48451
+run = CommandListEnvironment
+[ShaderOverridePalmTrees-20200610]
+; Bytecode hash: 112feb5a
+hash = ae68b2fe0f16091a
+run = CommandListEnvironment
+
+[CommandListEnvironment]

```

Comment:
Are there mods already relying on this in the wild? checktextureoverride adds some performance overhead every time it's called (doubly so if some mod is preventing post checktextureoverrides from being optimised out), so people running potato PCs would benefit from avoiding unecessary checktextureoverride calls. For most categories it's too late to change anything, but I had been thinking it might be worth adding a global variable to control whether these rarer categories get executed, and then mods that need them could set that variable to true in their initialisation command list.

If they are already in use I won't sweat it and am happy to pull this into the base mod - I can possibly add something into 3DMigoto itself to address any performance concerns, e.g. throttle these if none of the checktextureoverrides have matched anything since launch.

Reply:
So, there are now mods for better Palm Trees and holding the Sign-board and few other various things in the Owner room. Knight also included the shader for moving water but I'm not really sure for what mod of his. I do agree that likely adding another toggle or another option in the existing cycle for disabling them would be beneficial however I wasn't sure which route to go for.